src/controller/robot/controller.go

```diff
@@ -0,0 +1,316 @@
+package robot
+
+import (
+	"context"
+	"fmt"
+	rbac_common "github.com/goharbor/harbor/src/common/rbac"
+	"github.com/goharbor/harbor/src/common/utils"
+	"github.com/goharbor/harbor/src/core/config"
+	"github.com/goharbor/harbor/src/lib/errors"
+	"github.com/goharbor/harbor/src/lib/log"
+	"github.com/goharbor/harbor/src/lib/q"
+	"github.com/goharbor/harbor/src/pkg/permission/types"
+	"github.com/goharbor/harbor/src/pkg/project"
+	"github.com/goharbor/harbor/src/pkg/rbac"
+	rbac_model "github.com/goharbor/harbor/src/pkg/rbac/model"
+	robot "github.com/goharbor/harbor/src/pkg/robot2"
+	"github.com/goharbor/harbor/src/pkg/robot2/model"
+	"time"
+)
+
+var (
+	// Ctr is a global variable for the default robot account controller implementation
+	Ctr = NewController()
+)
+
+// Controller to handle the requests related with robot account
+type Controller interface {
+	// Get ...
+	Get(ctx context.Context, id int64, option *Option) (*Robot, error)
+
+	// Create ...
+	Create(ctx context.Context, r *Robot) (*Robot, error)
```

Comment:
Can we only return the id of the new created robot account? When the API handler want to know the secret of the created robot account, it can call the `Get` method of the controller.

Reply:
API will use the robot attribute to wrap a object and add it into the response body.

Reply:
API will use the robot attribute to wrap a object and add it into the response body.